pkg/addon/templateagent/template_agent.go

```diff
@@ -116,6 +117,9 @@ func (a *CRDTemplateAgentAddon) GetAgentAddonOptions() agent.AgentAddonOptions {
 			CSRApproveCheck:   a.TemplateCSRApproveCheckFunc(),
 			CSRSign:           a.TemplateCSRSignFunc(),
 		},
+		AgentDeployTriggerClusterFilter: func(old, new *clusterv1.ManagedCluster) bool {
+			return !equality.Semantic.DeepEqual(old.Annotations, new.Annotations)
```

Comment:
do you think we should only check the change of certain key? to reduce the reconcile call.

Reply:
fixed, PTAL